src/zenml/integrations/mlflow/model_deployers/mlflow_model_deployer.py

```diff
@@ -0,0 +1,226 @@
+        In the case of MLflow, we need to look into the local filesystem at a location
+        where the configuration for the service is stored. Retrieving that, we can
+        check if there's any service that was created for the existing model and 
+        can return that or create one if there isn't any. 
+        
+        Concrete model deployer subclasses must implement the following
+        functionality in this method:
+        - Detect if there is an existing model server instance running serving
+        one or more previous versions of the same model
+        - Deploy the model to the serving platform or update the existing model
+        server instance to include the new model version
+        - Return a Service object that is a representation of the external model
+        server instance. The Service must implement basic operational state
+        tracking and lifecycle management operations for the model server (e.g.
+        start, stop, etc.)
+
+        Args:
+            pipeline_name: Name of the pipeline that the model to be deployed
+                is part of.
+            run_id: ID of the pipeline run which the model to be deployed
+                is part of.
+            step_name: The name of the pipeline model deployment step that
+                deploys the model.
+            model_name: Name of the model to be deployed.
+            model_uri: URI of the model to be deployed.
+            model_type: Type/format of the model to be deployed.
+            config: Custom Service configuration parameters for the model
+                deployer.
+
+        Returns:
+            The deployment Service object.
+        """
+
+        services_path = os.path.join(
+            get_global_config_directory(),
+            LOCAL_STORES_DIRECTORY_NAME,
+            str(self.uuid),
+        )
+        
+        for root, dirs, files in os.walk(services_path):
+            for file in files:
+                if(file == SERVICE_DAEMON_CONFIG_FILE_NAME):
+                    service_config_path = os.path.join(root, file)                    
+                    logger.info(
+                        "Loading service daemon configuration from %s", service_config_path
+                    )
+                    old_config = None
+                    with open(service_config_path, "r") as f:
+                        old_config = f.read()
+                    mlflow_service = ServiceRegistry().load_service_from_json(old_config)
+                    if not isinstance(mlflow_service, MLFlowDeploymentService):
+                        raise TypeError(
+                            f"Expected service type LocalDaemonService but got "
+                            f"{type(mlflow_service)} instead"
+                        )
+                    if(self._old_service_exists(mlflow_service, config)):
+                        return mlflow_service.stop(timeout=10)
```

Comment:
If an old service exists, it needs to be stopped (if running) and, preferably, its configuration file/folder deleted from disk, otherwise they will clutter up the place. A new service always needs to be created, so you probably should not return here.

Another way to handle old services is to stop them (if running), then replace their configuration with the new one, then restart them. I haven't tried this before though, so there may be some issues with it.

Reply:
Next time we sync, we can talk about this too